pennylane/ops/qubit/hamiltonian.py

```diff
@@ -261,6 +261,34 @@ def grouping_indices(self):
         """
         return self._grouping_indices
 
+    @grouping_indices.setter
+    def grouping_indices(self, value):
+        """Set the grouping indices, if known without explicit computation, or if
+        computation was done externally. The groups are not verified.
+
+        **Example:**
+
+        Examples of valid groupings for the Hamiltonian
+
+        >>> H = qml.Hamiltonian([qml.PauliX('a'), qml.PauliX('b'), qml.PauliY('b')])
+
+        are
+
+        >>> H.grouping_indices = [[0, 1], [2]]
+
+        or
+
+        >>> H.grouping_indices = [[0, 2], [1]]
+
+        since both ``qml.PauliX('a'), qml.PauliX('b')`` and ``qml.PauliX('a'), qml.PauliY('b')`` commute.
+
+
+        Args:
+            value (list[list[int]]): List of lists of indexes of the observables in ``self.ops``. Each sublist
+                represents a group of commuting observables.
+        """
+        self._grouping_indices = value
+
```

Comment:
Added this setter so we do not have to compute groups if we know them!

Reply:
It's not checking for validity, but latest in the tape this will be done, and the error should be readable ("tried to measure non-commuting observables")

Reply:
Not needed for this PR, but it makes me realize the advantage of having a user-accessible way to specify 'this Hamiltonian is diagonal' to get the same effect!